src/software/ai/hl/stp/tactic/dribble/dribble_fsm.h

```diff
@@ -0,0 +1,220 @@
+#pragma once
+
+#include "shared/constants.h"
+#include "software/ai/evaluation/pass.h"
+#include "software/ai/hl/stp/tactic/move/move_fsm.h"
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/intent/move_intent.h"
+#include "software/geom/algorithms/acute_angle.h"
+#include "software/geom/algorithms/contains.h"
+#include "software/geom/algorithms/distance.h"
+
+struct DribbleFSM
+{
+    class GetPossessionState;
+    class DribbleState;
+
+    struct ControlParams
+    {
+        std::optional<Point> ball_destination;
+        std::optional<Angle> final_face_ball_oriention;
```

Comment:
Just to confirm, ball_destination is the point on the field where we want the robot to dribble the ball to, and final_face_ball_orientation is the orientation that the robot should rotate itself and the ball to?

Reply:
yes, do you have better name suggestions?

Reply:
Maybe ball_destination could be changed to dribble_destination, and final_face_ball_orientation could be changed to something like final_dribble_orientation? Something to indicate that the robot and ball are moving together